src/poetry/factory.py

```diff
@@ -126,10 +134,15 @@ def create_config(cls, io: Optional["IO"] = None) -> Config:
         return config
 
     @classmethod
-    def configure_sources(
-        cls, poetry: Poetry, sources: List[Dict[str, str]], config: Config, io: "IO"
-    ) -> None:
-        for source in sources:
+    def create_pool(cls, config: "Config", io: Optional["IO"] = None) -> "Pool":
+        from poetry.repositories.pool import Pool
+
+        if io is None:
```

Comment:
This feels a little unnecessary -- is it a big burden to require passing in `IO`? The caller could always provide a `NullIO`.